src/mad_spark_alt/core/robust_json_handler.py

```
@@ -0,0 +1,197 @@
+Robust JSON handling for LLM responses that might be malformed.
+"""
+
+import json
+import logging
+import re
+from typing import Any, Dict, List, Optional, Union
+
+logger = logging.getLogger(__name__)
+
+
+def extract_json_from_response(
```

Ruff (0.11.9)

`extract_json_from_response` is too complex (16 > 10)

(C901)